src/Playwright/Core/RawHeaders.cs

```diff
@@ -73,5 +73,11 @@ public string[] GetAll(string name)
 
             return null;
         }
+
+        internal static RawHeaders FromHeadersObjectLossy(IEnumerable<KeyValuePair<string, string>> headers)
+        {
+            var headersArray = headers.Select(x => new NameValue() { Name = x.Key, Value = x.Value }).ToList();
```

Comment:
You'll also want to filter out null `x.Value`'d items if you don't elsewhere to get the delete headers tests passing.